grc/core/generator/top_block.py

```diff
@@ -111,14 +112,17 @@ def _build_python_code_from_template(self):
             elif key == 'epy_module':
                 src = block.params['source_code'].get_value()
                 output.append((file_path, src))
+            elif not hasQtGui and "qtgui" in key:
```

Comment:
this is a bit redundant: the first condition `not hasQtGui` could be omitted.

Anyway, this is a bit *too* shallow for actually checking for whether a QtGui element exists in a flow graph. We're not forcing anyone to use "qtgui" in the names of their graphical sinks, and there's multiple out-of-tree visualization sinks that don't contain that string.

Reply:
That makes sense, I have only checked the in-tree blocks to see if they have something in common and that is how I arrived at name having "qtgui".

What about checking the python import statements to see if they contain `from PyQt5 import Qt`?

Reply:
Hm, that's a very hard question. Maybe we should be forcing people to explicitly declare their blocks as having QT windows, simply by having a new `flag` for the YAML files. That would make your life a lot easier here, wouldn't it?

Reply:
Qt blocks get added to the `top_grid_layout` in the flowgraph object in order to show up in the window. We can check the `count` of the layout to see if any blocks have been added to it and if not, hide the window. Unless someone is manually modifying the generated python, that should have all the included blocks.

Reply:
Having the explicit option to override and hide the window even if blocks exist would be nice too.